kale/loaddata/multiomics_datasets.py

```diff
@@ -0,0 +1,461 @@
+            dist = calculate_distance(train_data, test_data, eps=eps, metric=metric)
+            non_zero_entries = self._generate_sparse_adj(dist, self_loop=train)
+            adj_matrix[:num_train, num_train:] = torch.mul(dist, non_zero_entries)
+
+            dist = calculate_distance(test_data, train_data, eps=eps, metric=metric)
+            non_zero_entries = self._generate_sparse_adj(dist, self_loop=train)
+            adj_matrix[num_train:, :num_train] = torch.mul(dist, non_zero_entries)
+
+        adj_matrix_trans = adj_matrix.transpose(0, 1)
+        adj_matrix = (
+            adj_matrix
+            + adj_matrix_trans * (adj_matrix_trans > adj_matrix).float()
+            - adj_matrix * (adj_matrix_trans > adj_matrix).float()
+        )
+        identity_mat = torch.eye(adj_matrix.shape[0], device=adj_matrix.device)
+        adj_matrix = F.normalize(adj_matrix + identity_mat, p=1)
+        adj_matrix = adj_matrix.to_sparse()
+
+        return adj_matrix.indices(), adj_matrix.values()
+
+    def _find_sim_threshold(self, adj_mat: torch.Tensor, num_train: int) -> None:
+        r"""Finds a threshold for the adjacency matrix in order to keep the predefined number of edges per nodes.
+
+        Args:
+            adj_mat (torch.Tensor): The dense adjacency matrix.
+            num_train (int): The number of samples in training data.
+        """
+        sorted_adj_mat = torch.sort(adj_mat.reshape(-1,), descending=True).values[self.edge_per_node * num_train]
+        self.sim_threshold = sorted_adj_mat.item()
+
+    def _generate_sparse_adj(self, adj_mat: torch.Tensor, self_loop: bool = True) -> torch.Tensor:
+        r"""Returns a sparse adjacency matrix by setting entries below the ``sim_threshold`` to 0.
+
+        Args:
+            adj_mat (torch.Tensor): The dense adjacency matrix.
+            self_loop (bool, optional): Whether to fill the main diagonal with zero. (default: ``True``)
+
+        Returns:
+            torch.Tensor: Computed sparse adjacency matrix.
+        """
+        non_zero_entries = (adj_mat >= self.sim_threshold).float()
+        if self_loop:
+            non_zero_entries.fill_diagonal_(0)
+
+        return non_zero_entries
+
+    def _get_sample_weight(self, labels: np.ndarray) -> torch.Tensor:
+        r"""Get sample weights based on the class distribution.
+
+        Args:
+            labels (np.ndarray): A list of ground truth labels of samples.
+
+        Returns:
+            torch.Tensor: A list of label weights calculated for each sample.
+        """
+        if self.equal_weight:
+            sample_weight = np.ones(len(labels)) / len(labels)
+        else:
+            count = np.bincount(labels, minlength=self.num_classes)
+            sample_weight = count[labels] / np.sum(count)
+
+        sample_weight = torch.tensor(sample_weight, dtype=torch.float)
+
+        return sample_weight
+
+    def __repr__(self) -> str:
```

Comment:
A bit doc? Not obvious to me what it is for.

Reply:
I have replaced the __repr__ method with __str__ , as it provides a more user-friendly output for users (__repr__ is mostly used by developers). 
I have added a docstring for this method to provide clarity and explanation.